### PR TITLE
feat: add custom buttons for participant menu and toolbar via config

### PR DESCRIPTION
**Pull request by gabiborlea**
Screenshot 2023-01-26 at 15 27 00
Screenshot 2023-01-26 at 15 27 11

The buttons can be added by changing the config. Two config flags have been added: customParticipantMenuButtons and customToolbarButtons. Both of them are array of types: Array<{ icon: string; id: string; text: string; }>. Icon can be an URL or base64 encoding.
The buttons trigger API events: participantMenuButtonClick and toolbarButtonClicked

To test this feature use https://gborlea.jitsi.net/. Keep in mind to use the iFrameAPI with it and add the flags in configOverwrite.
Peace ✌️

### CODE REVIEW

File: react/features/base/ui/components/web/ContextMenu.tsx [hunk 1]
```diff
@@ -100,7 +100,7 @@ interface IProps {
     onMouseLeave?: (e?: React.MouseEvent) => void;
 }
 
-const MAX_HEIGHT = 400;
+const MAX_HEIGHT = 500;
```
Comment (saghul) [inline react/features/base/ui/components/web/ContextMenu.tsx:103]: Why do we need to change this height? I think we recently made adjustments to better accommodate small screens.

File: react/features/toolbox/components/web/Toolbox.tsx [hunk 2]
```diff
@@ -1524,11 +1546,11 @@ function _mapStateToProps(state: IReduxState, ownProps: Partial<IProps>) {
     const endConferenceSupported = conference?.isEndConferenceSupported() && isLocalParticipantModerator(state);
 
     const {
-        buttonsWithNotifyClick,
         callStatsID,
         disableProfile,
         iAmRecorder,
-        iAmSipGateway
+        iAmSipGateway,
+        customToolbarButtons
```
Comment (saghul) [inline react/features/toolbox/components/web/Toolbox.tsx:1553]: style: alpha sort

**Review by saghul — COMMENTED** [review]
Outstanding job! I left some minor comments, PTAL.

File: react/features/base/ui/components/web/ContextMenu.tsx (hunk 1 above)
Comment (gabiborlea) [reply react/features/base/ui/components/web/ContextMenu.tsx:103]: It remained like this when testing the feature. With 400 I had to scroll for the custom button added, so I set it up to 500

Comment (gabiborlea) [reply react/features/base/ui/components/web/ContextMenu.tsx:103]: I'll just set it back to 400

File: react/features/base/config/configWhitelist.ts [hunk 3]
```diff
@@ -78,6 +78,8 @@ export default [
     'channelLastN',
     'connectionIndicators',
     'constraints',
+    'customParticipantMenuButtons',
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/config/configWhitelist.ts:81]: Let's drop it for now.

File: react/features/video-menu/components/web/CustomOptionButton.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+const CustomOptionButton = (
+        { icon, onClick, text }:
+        {
+            icon: string;
+            onClick: (e?: React.MouseEvent<Element, MouseEvent> | undefined) => void;
+            text: string;
+    }
+) => {
+    const iconNode = useMemo(() => (<img
+        height = { 20 }
```
[the remark this answers is not in the document]
Comment (gabiborlea) [reply react/features/video-menu/components/web/CustomOptionButton.tsx:14]: I dont think so

[the remark this answers is not in the document]
Comment (saghul) [reply react/features/video-menu/components/web/CustomOptionButton.tsx:14]: HUm. Can you please check how the current buttons are sized?

File: react/features/base/config/configWhitelist.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gabiborlea) [reply react/features/base/config/configWhitelist.ts:81]: sure

**Review by saghul — APPROVED** [review]

**Comment by ledzepff** [comment]
Hi @saghul 

What happened with this feature? I've been looking to see if these changes have been merged into the master to the latest code there, but at least I cannot see them in external_api.js

**Comment by saghul** [comment]
Hey there! These changes landed a while ago already. What deployment are you using?

**Comment by ledzepff** [comment]
Yeah, my fault, shame on me ) BTW, we added a custom button, but from config.js, not through API, but as I understood, it has to be work via API, right?

**Comment by ledzepff** [comment]
Sorry, was my fault ) cheers

**Comment by conick** [comment]
Is there a way to make it visible or hidden via iframe api and is it possible to move it up in "three dots" menu?

**Pull request merged**